includes/class-wcpdf-main.php

```diff
@@ -257,12 +257,32 @@ public function get_documents_for_email( $email_id, $order ) {
 	 */
 	public function generate_pdf_ajax() {
 		$guest_access = isset( WPO_WCPDF()->settings->debug_settings['guest_access'] );
-		if ( !$guest_access && current_filter() == 'wp_ajax_nopriv_generate_wpo_wcpdf') {
+		if ( ! $guest_access && current_filter() == 'wp_ajax_nopriv_generate_wpo_wcpdf' ) {
+			wp_die( esc_attr__( 'You do not have sufficient permissions to access this page.', 'woocommerce-pdf-invoices-packing-slips' ) );
+		}
+
+		// check if the access type is 'order_key'
+		$order_key = false;
+		if ( ! empty( $_REQUEST['access_key'] ) ) {
+			if ( strpos( $_REQUEST['access_key'], 'wc_order_' ) !== false ) {
+				$order_key = true;
+			}
+		} else {
+			// handle legacy access keys
+			foreach ( array( '_wpnonce', 'order_key' ) as $legacy_key ) {
+				if ( ! empty( $_REQUEST[$legacy_key] ) ) {
+					$_REQUEST['access_key'] = sanitize_text_field( $_REQUEST[$legacy_key] );
+				}
+			}
+		}
+
+		// check if we have the access key
+		if ( empty( $_REQUEST['access_key'] ) ) {
 			wp_die( esc_attr__( 'You do not have sufficient permissions to access this page.', 'woocommerce-pdf-invoices-packing-slips' ) );
 		}
 
 		// Check the nonce - guest access doesn't use nonces but checks the unique order key (hash)
-		if( empty( $_REQUEST['action'] ) || ( !$guest_access && !check_admin_referer( $_REQUEST['action'] ) ) ) {
+		if ( empty( $_REQUEST['action'] ) || ( ! $order_key && ! check_admin_referer( $_REQUEST['action'], 'access_key' ) ) ) {
```

Comment:
You removed the check for `$guest_access` here, which means the setting no longer has any effect, right? (it will always be enabled as a result)

Reply:
Using the `shortcode` in a logged out store:

* Guest access == `true`
* Valid nonce? `No`

Dies here.

Reply:
If `$guest_access` is `true`, then `! $guest_access` will be `false`, so it won't die here.

Reply:
Yes, sorry, my brain got confused.